tests/remote_config/test_remote_configuration.py

```diff
@@ -134,15 +134,16 @@ def rc_check_request(data, expected, caching):
         ), f"targetsVersion was expected to be {expected_targets_version}, not {targets_version}"
 
         # verify that the tracer is properly storing and reporting on its config state
-        expected_config_states = client_state.get("config_states")
+        expected_config_states = expected.get("config_states")
```

Comment:
We were not using the `expected` here so we always compared the same client state